components/text.js

```diff
@@ -137,6 +137,80 @@ export default memo(function Text ({ rel, imgproxyUrls, children, tab, itemId, o
 
   const P = useCallback(({ children, node, ...props }) => <div className={styles.p} {...props}>{children}</div>, [])
 
+  const A = useCallback(({ node, href, children, ...props }) => {
+    children = children ? Array.isArray(children) ? children : [children] : []
+    // don't allow zoomable images to be wrapped in links
+    if (children.some(e => e?.props?.node?.tagName === 'img')) {
+      return <>{children}</>
+    }
+
+    // if outlawed, render the link as text
+    if (outlawed) {
+      return href
+    }
+
+    // If [text](url) was parsed as <a> and text is not empty and not a link itself,
+    // we don't render it as an image since it was probably a conscious choice to include text.
+    const text = children[0]
+    if (!!text && !/^https?:\/\//.test(text)) {
+      if (props['data-footnote-ref'] || typeof props['data-footnote-backref'] !== 'undefined') {
+        return (
+          <Link
+            {...props}
+            id={props.id && itemId ? `${props.id}-${itemId}` : props.id}
+            href={itemId ? `${href}-${itemId}` : href}
+          >{text}
+          </Link>
+        )
+      }
+      return (
+        // eslint-disable-next-line
+        <a id={props.id} target='_blank' rel={rel ?? UNKNOWN_LINK_REL} href={href}>{text}</a>
+      )
+    }
+
+    try {
+      const linkText = parseInternalLinks(href)
+      if (linkText) {
+        return <a target='_blank' href={href} rel='noreferrer'>{linkText}</a>
+      }
+    } catch {
+      // ignore errors like invalid URLs
+    }
+
+    // if the link is to a youtube video, render the video
+    const youtube = href.match(/(https?:\/\/)?((www\.)?(youtube(-nocookie)?|youtube.googleapis)\.com.*(v\/|v=|vi=|vi\/|e\/|embed\/|user\/.*\/u\/\d+\/)|youtu\.be\/)(?<id>[_0-9a-z-]+)((?:\?|&)(?:t|start)=(?<start>\d+))?/i)
+    if (youtube?.groups?.id) {
+      return (
+        <div style={{ maxWidth: topLevel ? '640px' : '320px', paddingRight: '15px', margin: '0.5rem 0' }}>
+          <YouTube
+            videoId={youtube.groups.id} className={styles.youtubeContainer} opts={{
+              playerVars: {
+                start: youtube?.groups?.start
+              }
+            }}
+          />
+        </div>
+      )
+    }
+
+    // if the link is to a rumble video, render the video
+    const rumble = href.match(/(?:https?:\/\/)?(?:www\.)?rumble\.com\/embed\/(?<id>[a-z0-9]+)\/\?pub=\d+&start=(?<start>\d+)/i)
```

Comment:
I tested with this URL: https://rumble.com/embed/v17j3tt/?pub=3fid3f

As you can see, `pub` can be alphanumeric so you want to use `[a-z0-9]+` or `\w+` for it. And  `start` should not be required to match.

I noticed that we could probably leverage the URL constructor here for parsing [as I did in `parseInternalLinks`](https://github.com/stackernews/stacker.news/blob/b4cef44a43bcdb802e1d035748c84a420aa22218/lib/url.js#L29-L52) (see https://github.com/stackernews/stacker.news/pull/765#issuecomment-1907650129). I think this would make this easier and cleaner.

The other twitter and youtube embed code you can see here is just old and bad and isn't using them yet.